### PR TITLE
[terra-functional-testing] Add documentation for environment variables

### LINKED ISSUE #528: [terra-functional-testing] Add documentation for environment variables
# Story

## Description

As a consumer of terra I want to create documentation for each of the environment variables used within the terra-functional-testing library.

## Motivation

So that consumers have detailed documentation for all ENVs that can alter the behavior of the test run.

## Definition of Done

We'll know we're done when a PR has been merged to main that contains descriptions for each of the ENV variables.

### PR DESCRIPTION
**Pull request by StephenEsser**
### Summary

This PR adds documentation for the environment variables that are used during the test run.

Resolves #528

Deployment:

https://terra-toolki-story-528-jwtrwre.herokuapp.com/dev_tools/terra-functional-testing/terra-functional-testing/wdio-testing/about#environment-variables

@cerner/terra

[CONTRIBUTORS.md]: ../blob/main/CONTRIBUTORS.md
[License]: ../blob/main/LICENSE

### CODE REVIEW

File: packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx [hunk 1]
```diff
@@ -59,6 +57,24 @@ Tests can be executed in any of the following supported form factors:
...
+| FORM_FACTOR                   | string  | true     |           | A form factor assigned to the test run.                                                                                                   |
+| LOCALE                        | string  | true     |           | A locale assigned to the test run.                                                                                                        |
+| SELENIUM_GRID_URL             | string  | true     |           | The remote selenium grid address.                                                                                                         |
+| SITE                          | string  |          |           | A file path to a static directory of assets. When defined, an express server will launch to serve the assets and disable running webpack. |
+| THEME                         | string  | true     |           | A theme name assigned to the test run.                                                                                                    |
```
Comment (emilyrohrbough) [inline packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]: This default value is `terra-default-theme` correct?

File: packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx [hunk 2]
```diff
@@ -59,6 +57,24 @@ Tests can be executed in any of the following supported form factors:
 | huge     | 1300  | 768    |
 | enormous | 1500  | 768    |
 
+## Environment Variables
+
+The following environment variables are configurable for the test run. Variables marked as reserved will be overwritten by the test runner. To configure reserved variables use the test runner configuration options. If the test runner is not being used the variables are configurable by the user.
+
+| Variable Name                 | Type    | Reserved | Default   | Description                                                                                                                               |
+|-------------------------------|---------|----------|-----------|-------------------------------------------------------------------------------------------------------------------------------------------|
+| BROWSERS                      | string  | true     |           | A list of browsers for the test run. Browsers are delimited by a comma. Example: chrome,firefox,ie                                        |
```
Comment (emilyrohrbough) [inline packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]: Technically by default this is chrome & firefox correct?

File: packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx [hunk 3]
```diff
@@ -59,6 +57,24 @@ Tests can be executed in any of the following supported form factors:
...
+| FORM_FACTOR                   | string  | true     |           | A form factor assigned to the test run.                                                                                                   |
+| LOCALE                        | string  | true     |           | A locale assigned to the test run.                                                                                                        |
```
Comment (emilyrohrbough) [inline packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]:
```suggestion
| LOCALE                        | string  | true     |   'en'        | A locale assigned to the test run.                                                                                                        |
```

File: packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx (hunk 1 above)
Comment (StephenEsser) [reply packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]: Hmm, kind of. There's technically not a default if you don't use the test runner but the services do implement a default so I think it makes sense to add it here. We might need to tweak a few things to make sure this happens as expected everywhere a theme is referenced

File: packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx (hunk 2 above)
Comment (StephenEsser) [reply packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]: There's two defaults, locally (not on the grid) it's only chrome, if the grid is enabled it's chrome, firefox, and ie

File: packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx (hunk 3 above)
Comment (StephenEsser) [reply packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]: Added en as the default:

https://github.com/cerner/terra-toolkit/pull/544/commits/545286169a5e7ce61f25b37444a1105be829b750

File: packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx (hunk 1 above)
Comment (StephenEsser) [reply packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]: Added terra-default-theme as the default:

https://github.com/cerner/terra-toolkit/pull/544/commits/545286169a5e7ce61f25b37444a1105be829b750

File: packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx (hunk 2 above)
Comment (StephenEsser) [reply packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]: I added chrome as the default:

https://github.com/cerner/terra-toolkit/pull/544/commits/545286169a5e7ce61f25b37444a1105be829b750

**Comment by StephenEsser** [comment]
The table is visible on the deployed site if anyone wants to see how it looks to a user viewing the documentation:

https://terra-toolki-story-528-jwtrwre.herokuapp.com/dev_tools/terra-functional-testing/terra-functional-testing/wdio-testing/about#environment-variables

File: packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx [hunk 4]
```diff
@@ -59,6 +57,24 @@ Tests can be executed in any of the following supported form factors:
...
+## Environment Variables
+
+The following environment variables are configurable for the test run. Variables marked as reserved will be overwritten by the test runner. To configure reserved variables use the test runner configuration options. If the test runner is not being used the variables are configurable by the user.
+
+| Variable Name                 | Type    | Reserved | Default             | Description                                                                                                                               |
+|-------------------------------|---------|----------|---------------------|-------------------------------------------------------------------------------------------------------------------------------------------|
+| BROWSERS                      | string  | true     | chrome              | A list of browsers for the test run. Browsers are delimited by a comma. Example: chrome,firefox,ie                                        |
+| FORM_FACTOR                   | string  | true     |                     | A form factor assigned to the test run.                                                                                                   |
```
Comment (benbcai) [inline packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]: Do we also want to call out that the default is `huge`? https://github.com/cerner/terra-toolkit/blob/main/packages/terra-functional-testing/src/services/wdio-terra-service.js#L99

I don't know, what do you think?

File: packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx [hunk 5]
```diff
@@ -59,6 +57,24 @@ Tests can be executed in any of the following supported form factors:
...
+| LOCALE                        | string  | true     | en                  | A locale assigned to the test run.                                                                                                        |
+| SELENIUM_GRID_URL             | string  | true     |                     | The remote selenium grid address.                                                                                                         |
+| SITE                          | string  |          |                     | A file path to a static directory of assets. When defined, an express server will launch to serve the assets and disable running webpack. |
+| THEME                         | string  | true     | terra-default-theme | A theme name assigned to the test run.                                                                                                    |
+| WDIO_DISABLE_SELENIUM_SERVICE | boolean |          |                     | A flag to disable the selenium docker service.                                                                                            |
```
Comment (benbcai) [inline packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]: Should we call out the default for WDIO_DISABLE_SELENIUM_SERVICE is `false`?

File: packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx (hunk 4 above)
Comment (StephenEsser) [reply packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]: Hmm, this is a tricky one. It's a bit of a gray area. When a FORM_FACTOR is set only that form factor is allowed to run. When no FORM_FACTOR is set we loop through all form factors using the describeViewports utility.

File: packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx (hunk 5 above)
Comment (StephenEsser) [reply packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]: I think that makes sense. I overthink some of these a little too much, but for all intents and purposes this is false (even though it's actually undefined/not set). I'll update this to false

Comment (StephenEsser) [reply packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]: Update to default to false:

https://github.com/cerner/terra-toolkit/pull/544/commits/eb863d146f8d4cdc9ac5c7e661bd2830bf548a8f

File: packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx (hunk 4 above)
Comment (StephenEsser) [reply packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]: I went ahead and specified this as `huge`:

https://github.com/cerner/terra-toolkit/pull/544/commits/eb863d146f8d4cdc9ac5c7e661bd2830bf548a8f

**Review by mjhenkes — APPROVED** [review]

File: packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx (hunk 4 above)
Comment (benbcai) [reply packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]: Or should we leave it blank and just update the description to include what you just said and that only if that form_factor is one of the form factors listed in describeViewports?

Comment (benbcai) [reply packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]: Ok. that's fine. I didn't see your latest comment before adding my latest comment.

**Review by benbcai — APPROVED** [review]

**Pull request merged**